src/main/java/com/apeshko/javaspring/storage/Storage.java

```diff
@@ -0,0 +1,10 @@
+package com.apeshko.javaspring.storage;
+
+import java.util.List;
+
+public interface Storage {
```

Comment:
to avoid class casting, better to create generic interface like with repositories
public interface Storage<T> {
    T getByKey(String key);
    List<T> getAllByKeyPrefix(String keyPrefix);
    T save(String key, T value);
    T remove(String key);
}